### DISCUSSION TITLE
x-bind classes as an object to allow conditions

### DISCUSSION THREAD

**Discussion by ghost** [alpinejs/alpine#2817] [3. Feature ideas]
It would be very useful to allow binding classes conditionally, something like:

```
<div :class="{
        'always-present' : true,
        'has-errors' : isInvalid,
        'invalid-classname' : isDisabled,
        'whatever-only-when-no-issues' : !isInvalid && !isDisabled,
 }">
```

I am currently doing this with a custom function that updates a variable called `allClasses` with `$watch` and then `:class="allClasses"`

**Comment by DiskoPete** [comment]
Isn't this already possible?
https://alpinejs.dev/directives/bind#class-object-syntax
